### PR TITLE
feat: Add TSEditQuery

### PR DESCRIPTION
**Pull request by theHamsta**
~~This is currently affected by the runtime file duplication bug:
https://github.com/neovim/neovim/issues/14294~~

EDIT: used the upstream method that does the dedup and include inherited languages.

### CODE REVIEW

File: lua/nvim-treesitter/configs.lua [hunk 1]
```diff
@@ -245,6 +265,13 @@ M.commands = {
       "-nargs=0",
     },
   },
+  TSEditQuery = {
+    run = M.edit_query_file,
+    args = {
+      "-nargs=+",
+      "-complete=custom,nvim_treesitter#available_query_groups",
+    },
+  },
```
Comment (theHamsta) [inline lua/nvim-treesitter/configs.lua:274]: Maybe a version of this command to add/edit a query extension in the users config directory?

Comment (vigoux) [reply lua/nvim-treesitter/configs.lua:279]: I think the default version of the command should ask if the user wants to create a new extension if it does not already exists.

File: lua/nvim-treesitter/configs.lua [hunk 2]
```diff
@@ -1,6 +1,7 @@
 local api = vim.api
 
 local queries = require'nvim-treesitter.query'
+local nvim_query = require'vim.treesitter.query'
```
Comment (vigoux) [inline lua/nvim-treesitter/configs.lua:4]: WTF we need clearer names

File: lua/nvim-treesitter/configs.lua [hunk 3]
```diff
@@ -196,6 +197,30 @@ local function config_info(process_function)
...
+      vim.cmd(':edit '..files[choice])
+    end
+  end
+end
+
+function M.edit_query_file_user_after(query_group, lang)
+  lang = lang or parsers.get_buf_lang()
+  local folder = utils.join_path(vim.fn.stdpath('config'), 'after', 'queries', lang)
+  local file = utils.join_path(folder, query_group..'.scm')
+  pcall(vim.fn.mkdir, folder, "p", "0755")
```
Comment (vigoux) [inline lua/nvim-treesitter/configs.lua:220]: Error report here ?

File: lua/nvim-treesitter/query.lua [hunk 4]
```diff
@@ -23,6 +23,19 @@ for _, query in ipairs(M.built_in_query_groups) do
...
+  for _, f in ipairs(query_files) do
+    groups[vim.fn.fnamemodify(f, ':t:r')] = true
+  end
+  local list = {}
+  for k, _ in pairs(groups) do
+    table.insert(list, k)
+  end
+  return list
+end
+
```
Comment (vigoux) [inline lua/nvim-treesitter/query.lua:38]: Shouldn't we just use the default query groups ?

File: lua/nvim-treesitter/configs.lua (hunk 1 above)
Comment (theHamsta) [reply lua/nvim-treesitter/configs.lua:287]: Edit does not really create a file. So I think we should only ask for the folder.

File: lua/nvim-treesitter/query.lua (hunk 4 above)
Comment (theHamsta) [reply lua/nvim-treesitter/query.lua:38]: Why? It works for all query groups of all plugins.

Comment (vigoux) [reply lua/nvim-treesitter/query.lua:38]: Yup, I just thought about that after hitting enter, it sounds better this way.

**Review by vigoux — APPROVED** [review]

**Pull request merged**